src/Backups/IBackupCoordination.h

```diff
@@ -22,6 +22,7 @@ class IBackupCoordination
 
     /// Sets the current stage and waits for other hosts to come to this stage too.
     virtual void setStage(const String & new_stage, const String & message) = 0;
+    virtual void setStageForCluster(const String & new_stage) = 0;
```

Comment:
What is the "stage for cluster"? Each host has its own stage.

Reply:
When we check for concurrency and allow a backup/restore to proceed, we need to inform zookeeper that this backup/restore is going to be processed even before each host gets the query. As mentioned in above comment, here having a state is needed.